dashboard/src/actions/overviewActions.js

```diff
@@ -156,17 +147,18 @@ export const updateDataset =
         );
 
         for (const key in response.data.metadata) {
-          if (checkNestedPath(key, item.metadata) || key in item.metadata) {
-            if (checkNestedPath(key, item.metadata)) {
-              item.metadata = setValueFromPath(
-                key,
-                item.metadata,
-                response.data.metadata[key]
-              );
-            }
-            if (key in item.metadata) {
-              item.metadata[key] = response.data.metadata[key];
-            }
+          if (checkNestedPath(key, item.metadata)) {
+            item.metadata = setValueFromPath(
+              key,
+              item.metadata,
+              response.data.metadata[key]
+            );
+          } else if (item.metadata[key.split(".")[0]] === null) {
+            assignToNull(
+              item.metadata,
+              key.split("."),
+              response.data.metadata[key]
+            );
           }
```

Comment:
I would like to suggest an alternative approach:  rather than checking if the key is in the nested path and then calling a separate routine if it's not, let's enhance `setValueFromPath()` to handle the case.  I think that this is easily done:
```javascript
const setValueFromPath = (path, obj, value) => {
  const recurse = (plist, o, v) => {
    const [head, ...rest] = plist;
    return { ...o, [head]: rest.length ? recurse(rest, o[head] ?? {}, v) : v };
  };
  return recurse(path.split("."), obj, value);
};
```
That is, all I did was add a default case to the inner `recurse()` call to add an empty object if one doesn't exist.

And then the code here collapses to just this:
```javascript
        for (const key in response.data.metadata) {
          item.metadata = setValueFromPath(
            key,
            item.metadata,
            response.data.metadata[key]
          );
```
I think....  (Why is there an `else if` at line 156 with no final `else`?...are we missing a case, here??)

Reply:
I'll go with as how it is now. It  falls in either of them

Reply:
> I'll go with as how it is now.

😛  That means more code.

> It falls in either of them

If control really always executes either `setValueFromPath()` or `assignToNull()`, then we don't need the `if` at line 156 -- that can just be an `else`.  (I'm concerned that there might be a case where both `if` statements fail and we don't assign the metadata.)